### DISCUSSION TITLE
What causes (and how to avoid) sheering when scaling a compound shape

### DISCUSSION THREAD

**Discussion by DennisSmolek** [jrouwe/JoltPhysics.js#173] [Q&A] [answered]
[In the docs](https://jrouwe.github.io/JoltPhysics/class_sphere_shape.html#a94316a5849ea7c0bdbd55b9cf9da1195) it says:
> compound shapes cannot handle shapes being rotated and scaled (this would cause shearing),

I kinda get the concept of sheering and could expect some sheering with layered non-uniform scaling but I'm trying to understand what can cause it (so I can protect against it)

Does this mean I can't scale a compound shape at all?

Or that child shapes in a compound shape can't be scaled?

Or the way I'm reading it, scaling a compound shape that also has scaled/rotated shapes must be uniformly scaled. 

Is that correct?

**Comment by jrouwe** [comment, marked as answer]
* Uniform scale is always ok
* Non-uniform scale causes problems if a shape is first rotated under angles other than 90 degrees increments

See:

https://github.com/jrouwe/JoltPhysics/blob/343687a61493b1f0642b97b4b7fd05484acc3048/Jolt/Physics/Collision/Shape/CompoundShape.h#L188-L195

and

https://github.com/jrouwe/JoltPhysics/blob/343687a61493b1f0642b97b4b7fd05484acc3048/Jolt/Physics/Collision/Shape/ScaleHelpers.h#L36C17-L54
